### ISSUE TITLE
how can intialize my custom header view elegantly?

### ISSUE THREAD

**Issue by hjw541988478** [cymcsg/UltimateRecyclerView#132]
how should i do ? my headerview layout is quite complex.when i intialize these views , how can i populate the data ?

**Comment by sergii-frost** [comment]
Hi @hjw541988478 

your `CustomerAdapter` class should subclass `UltimateViewAdapter` which implements `StickyRecyclerHeadersAdapter` interface which has methods like: 

```
long getHeaderId(int var1);

VH onCreateHeaderViewHolder(ViewGroup var1);

void onBindHeaderViewHolder(VH var1, int var2);
```

Your implementation may look like: 

```
        @Override
        public RecyclerView.ViewHolder onCreateHeaderViewHolder(ViewGroup viewGroup) {
            View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.your_header_layout, viewGroup, false);
            return new YourHeaderViewHolder(v);
        }

        @Override
        public void onBindHeaderViewHolder(RecyclerView.ViewHolder holder, int position) {
            //Need to check that viewHolder is of expected class and there is data for corresponding position
            //As when using Sticky Headers and Load More, there can be crash
            //See https://github.com/cymcsg/UltimateRecyclerView/issues/59
            if (holder instanceof YourHeaderViewHolder) {
                //init your header subviews with values based on position
            }
        }
...
public class YourHeaderViewHolder extends UltimateRecyclerviewViewHolder {

            //Your header subviews here e.g.
            TextView headerTitle;

            public YourHeaderViewHolder(View itemView) {
                super(itemView);

                headerTitle = (TextView) itemView.findViewById(R.id.yourHeaderTitleId);
            }
        }
```

If you need to handle click on your header, you can implement `StickyRecyclerHeadersTouchListener.OnHeaderClickListener` interface in your Adapter class, and implement `onHeaderClick` method.

**Comment by hjw541988478** [comment]
@sergii-frost works fine,but how can i handle click event on exact child views on header view ?

**Comment by sergii-frost** [comment]
@hjw541988478 there is an issue for that specific case: https://github.com/cymcsg/UltimateRecyclerView/issues/123 
we need to wait for it to be fixed unfortunately. I haven't found any workaround for it yet.

**Comment by hjw541988478** [comment]
@sergii-frost thx anyhow.and can u fix my another issue #142 ,it bothers me a lot.

**Comment by sergii-frost** [comment]
@hjw541988478 I am not developing this library :) I am same user as you
